### PR TITLE
Move to ahesh for known keys as suggested in #290

### PR DESCRIPTION
**Pull request by Licenser**
@ritchie46 want to take a look and see if / how this works for you?

### CODE REVIEW

**Comment by ritchie46** [comment]
Yes! I shall take a look! Thanks already.

File: Cargo.toml [hunk 1]
```diff
@@ -18,9 +18,14 @@ features = ["docsrs"]
...
 beef = { version = "0.5", optional = true }
 halfbrown = "0.2"
-value-trait = { version = "0.6" }
+value-trait = { version = "0.6.1" }
 simdutf8 = { version = "0.1.4", features = ["public_imp", "aarch64_neon"] }
 
+# ahash known key
+once_cell = { version = "1.17", optional = true }
+ahash = { version = "0.8", optional = true }
+rand = { version = "0.8", optional = true }
```
Comment (ritchie46) [inline Cargo.toml:27]: Why is `rand` needed? Couldn't find any usage of it in this PR.

**Review by ritchie46 — COMMENTED** [review]
I went through the whole PR and I think this would work. :+1: 

With the `ObjectHasher` it seems abstracted away nicely.

**Comment by Licenser** [comment]
I'll try to release this as 0.10 today (as the changes in hasher might be breaking backwards compatibility)

**Pull request merged**